src/Components/Components/ref/Microsoft.AspNetCore.Components.netstandard2.0.cs

```diff
@@ -306,6 +333,15 @@ public static partial class HttpClientJsonExtensions
         [System.Diagnostics.DebuggerStepThroughAttribute]
         public static System.Threading.Tasks.Task<T> SendJsonAsync<T>(this System.Net.Http.HttpClient httpClient, System.Net.Http.HttpMethod method, string requestUri, object content) { throw null; }
     }
+    public partial interface IAuthenticationState
```

Comment:
Can we turn this into a sealed class? This just seems like homework for someone to have to implement, it's just data and we want to keep it that.

The provider already has to be stateful, so there's really no reason why you'd want to attach extra state to this type.

Reply:
I'm fine with turning it into a class, but marking it as `sealed` seems pretty harsh. You might want to attach extra state, for example to hold authentication tokens that were returned by the server when this is used on the client.

I know there are other theoretically possible ways to attach extra state (`WeakDictionary`, or embedding them in the claims) but I would think we can make it easier by subclassing.

Reply:
I was thinking of making it a class before, because I wanted to add some convenience properties, such as a top-level `bool IsAuthenticated` and `string Name` that got their data from the default identity. It's much more obvious and pleasant to write `AuthState.IsAuthenticated` than `AuthState.User.Identity.IsAuthenticated`.

I won't add the convenience properties just now because I'd like this feature and its usage patterns to bake a bit more first. It's something we can consider later by the time we're making templates and samples.

Reply:
Changed it to be a class.

@rynowak Let me know if you disagree about leaving it unsealed.

Reply:
I don't feel super strongly about being unsealed. 

> You might want to attach extra state, for example to hold authentication tokens that were returned by the server when this is used on the client.

I think that people are going to ask for this one way or another, but the current design really neither helps nor harms it. I think it's fine to address this in a future release.